external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -3370,12 +3376,17 @@ fn parse_body(context: &mut Context, native: Option<Loc>) -> Result<FunctionBody
                     ))),
                 )
             };
-
             let end_loc = context.tokens.previous_end_loc();
-            Ok(sp(
-                make_loc(context.tokens.file_hash(), start_loc, end_loc),
-                FunctionBody_::Defined(seq),
-            ))
+            let loc = make_loc(context.tokens.file_hash(), start_loc, end_loc);
+            if interface_only && !is_macro {
```

Comment:
I understand not wanting to parse the body when not needed, but this feels a bit strange since you are actually parsing it, just throwing it away.

Reply:
Yeah, I was not sure where to plug it in (not remembering that we still have a similar mechanism), and also hoped that we can skip parsing altogether at some point perhaps. In any case, now that it's rewritten with the existing machinery in mind, everything happens in expansion